bchrpc/bchrpc.proto

```diff
@@ -68,19 +70,44 @@ service bchrpc {
     // for the given address.
     //
     // **Requires AddressIndex**
+    // **Requires SlpIndex for slp related information **
     rpc GetAddressUnspentOutputs(GetAddressUnspentOutputsRequest) returns (GetAddressUnspentOutputsResponse) {}
 
     // GetUnspentOutput takes an unspent output in the utxo set and returns
     // the utxo metadata or not found.
+    //
+    // **Requires SlpIndex for slp related information **
     rpc GetUnspentOutput(GetUnspentOutputRequest) returns (GetUnspentOutputResponse) {}
 
     // GetMerkleProof returns a Merkle (SPV) proof for a specific transaction
     // in the provided block.
     //
-    // **Requires TxIndex***
+    // **Requires TxIndex**
     rpc GetMerkleProof(GetMerkleProofRequest) returns (GetMerkleProofResponse) {}
 
-    // SubmitTransaction broadcasts a transaction to all connected peers.
+    // GetTokenMetadata return SLP token metadata for one or more tokens.
+    //
+    // **Requires SlpIndex**
+    rpc GetTokenMetadata(GetTokenMetadataRequest) returns (GetTokenMetadataResponse) {}
+
+    // GetParsedSlpScript returns marshalled object from parsing an SLP pubKeyScript 
+    // using goslp package.  This endpoint does not require SlpIndex.
+    rpc GetParsedSlpScript(GetParsedSlpScriptRequest) returns (GetParsedSlpScriptResponse) {}
+
+    // GetTrustedValidation returns SLP validity related information for one or more transactions.
+    //
+    // **Requires SlpIndex**
+    rpc GetTrustedSlpValidation(GetTrustedSlpValidationRequest) returns (GetTrustedSlpValidationResponse) {}
+
+    // GetBip44HdAddress returns an address associated with a provided xpub, address index, and
+    // selecting either internal or external by setting the change boolean.  If SlpIndex is disabled
+    // the slpAddress string will be an empty string.
+    rpc GetBip44HdAddress(GetBip44HdAddressRequest) returns (GetBip44HdAddressResponse) {}
```

Comment:
Do we really need this one? It seems like this is just exposing a library function rather than accessing any unique data that the server has. Shouldn't this be done client side?

Reply:
I don't mind this being there, but do agree with Chris's point. These types of functions seem better for bchutil, rather than the gRPC implementation.

Reply:
Ok, will remove.

Reply:
This is now removed.